libraries/RC_Channel/RC_Channel.cpp

```diff
@@ -407,32 +412,38 @@ void RC_Channel::read_mode_switch()
     else if (pulsewidth < 1750) position = 4;
     else position = 5;
 
-    if (mode_switch_state.last_position == position) {
-        // nothing to do
-        return;
-    }
-
-    const uint32_t tnow_ms = AP_HAL::millis();
-    if (position != mode_switch_state.debounced_position) {
-        mode_switch_state.debounced_position = position;
-        // store time that switch last moved
-        mode_switch_state.last_edge_time_ms = tnow_ms;
-        return;
-    }
-
-    if (tnow_ms - mode_switch_state.last_edge_time_ms < MODE_SWITCH_DEBOUNCE_TIME_MS) {
-        // still in debounce
+    if (!debounce_completed(position)) {
         return;
     }
 
     // set flight mode and simple mode setting
     mode_switch_changed(position);
+}
+
+bool RC_Channel::debounce_completed(int8_t position) 
+{
+    // switch change not detected
+    if (switch_state.current_position == position) {
+        // reset debouncing
+         switch_state.debounce_position= position;
+    } 
+    // switch change detected
+    else {
```

Comment:
The else should be on the same as the closing bracket, the comment cam be put in the line below.

Reply:
fixed